spec/grape/endpoint_spec.rb

```diff
@@ -694,17 +694,6 @@ def app
         end.to raise_error(NoMethodError, %r{^undefined method `undefined_helper' for #<Class:0x[0-9a-fA-F]+> in `/hey' endpoint})
       end
     end
-
-    context 'when performing an undefined method of an instance inside the API' do
-      it 'raises NoMethodError but stripping the internals of the Object class' do
-        subject.get('/hey') do
-          Object.new.x
-        end
-        expect do
-          get '/hey'
-        end.to raise_error(NoMethodError, /^undefined method `x' for #<Object:0x[0-9a-fA-F]+>$/)
-      end
-    end
   end
```

Comment:
What's the reason for removing this? It was fixing a bug I believe with failure context.

Reply:
Well, it works for the `endpoint` through the`method_missing` but otherwise, the `NoMethodError` is handled by the object it self. Also when you look at the code, its supposed to end with `in `/hey' endpoint` but it does not. We don't have the control over all the instances declared in the endpoint but we do at the endpoint's instance level.